### ISSUE TITLE
bug: Android Capacitor Http Issue "Attempt to invoke virtual method ... getClass"

### ISSUE THREAD

**Issue by h3786010** [ionic-team/capacitor#6479]
### Capacitor Version

```
@capacitor/cli: 4.7.3
@capacitor/core: 4.7.3
@capacitor/android: 4.7.3
@capacitor/ios: 4.7.3
```

### Platform(s)

Android

### Current Behavior

First using Capacitor Http request return the error after launch app
![Screenshot_2023-04-06-17-13-02-61_67ee557a188fd8d45d3ec93c08900d76](https://user-images.githubusercontent.com/4884867/230812246-3718e70c-5ee6-4c87-a18a-279a040755fe.jpg)

### Code Reproduction

https://github.com/h3786010/capacitor-http-bug-repro

App.vue
```
<script setup>
import { onMounted } from 'vue'
import { CapacitorHttp } from '@capacitor/core'

onMounted(async () => {

  await CapacitorHttp.post({
    url: 'https://mydomain/api/foo',
    headers: {
      "Content-Type": "application/json",
    },
    webFetchExtra: {
      credentials: "include",
    },
  }).then((response) => {
    alert(JSON.stringify(response.data));
  }).catch((error) => {
    alert(error.message);
  })

})
</script>

<template>
  hello world!
</template>
```

capacitor.config.json
```
{
  "appId": "com.example.app",
  "appName": "test",
  "webDir": "dist",
  "bundledWebRuntime": false,
  "server": {
    "allowNavigation": [
      "mydomain"
    ]
  },
  "plugins": {
    "CapacitorHttp": {
      "enabled": true
    }
  }
}
```

### Other Technical Details

`npm --version` 8.19.3

`node --version` v18.13.0

`pod --version` output (iOS issues only):

**Comment by Vansinnesvisor** [comment]
Can confirm, POST requests not working. This issue broke my app release in Android.
Works well with server property in capacitor.config.ts (dev mode).
Currently rollback to 4.7.2.

**Comment by elenche** [comment]
Same issue here, tested on `@capacitor/android` versions `4.7.1` and `4.7.3`, both on device and simulator. It happens **only on the first app installation**. On subsequent opens the POST calls work as expected.

@zarko-tg  Downgrading to `@capacitor/android@4.7.2` solved the issue for me.

```
$ ionic info

Ionic:
   Ionic CLI                     : 6.16.3 (C:\Users\xyz\AppData\Roaming\npm\node_modules\@ionic\cli)
   Ionic Framework               : @ionic/angular 6.7.1
   @angular-devkit/build-angular : 14.2.11
   @angular-devkit/schematics    : 14.2.11
   @angular/cli                  : 14.2.11
   @ionic/angular-toolkit        : 6.1.0

Capacitor:
   Capacitor CLI      : 4.7.3
   @capacitor/android : 4.7.2
   @capacitor/core    : 4.7.3
   @capacitor/ios     : 4.7.3

Utility:
   cordova-res (update available: 0.15.4) : 0.15.3
   native-run                             : 1.7.2

System:
   NodeJS : v18.12.1 (C:\Program Files\nodejs\node.exe)
   npm    : 8.19.2
   OS     : Windows 10
```

**Comment by adnbrownie** [comment]
Getting the same stack trace in Android Logcat as in this comment: https://github.com/ionic-team/capacitor/issues/6479#issuecomment-1501346090

This happens in v4.7.3, v4.7.2 is fine. Changes between both versions: https://github.com/ionic-team/capacitor/compare/4.7.2...4.7.3

@jcesarmobile, could you elaborate why you set the https://github.com/ionic-team/capacitor/labels/needs%20reproduction label (i.e. what kind of additional input information you require)? The original issue report contains a link to this repo: https://github.com/h3786010/capacitor-http-bug-repro

**Comment by jcesarmobile** [comment]
Sorry, I missed the link, but sadly, I've tested it now and can't reproduce the issue on the provided app.

A customer has reported a similar issue with a proper reproduction and a coworker is looking into it, so if it turns out being the same issue, it will be fixed soon, but would help if we had a proper reproduction in this issue too so we can verify it's the same issue.

**Comment by prajapatijay95** [comment]
@ItsChaceD Can you please deploy the new dev version of 4.x branch for this changes?

**Comment by h3786010** [comment]
Working well after upgrade 4.8.0
Thank You!

need additional settings CapacitorCookies in my app project
capacitor.config.json
```
"plugins": {
  "CapacitorHttp": {
    "enabled": true
  }
}
```
to
```
"plugins": {
  "CapacitorCookies": {
    "enabled": true
  },
  "CapacitorHttp": {
    "enabled": true
  }
}
```

**Comment by adnbrownie** [comment]
I can confirm that the described issue is no longer happening with 4.8.0 when having both `CapacitorHttp` & `CapacitorCookie` enabled in `capacitor.config.json`, as @h3786010 described. Enabling only `CapacitorHttp` alone does not work, though.

**Comment by h3786010** [comment]
@adnbrownie Because my APP works through Session, Maybe CapacitorCookie needs to be explicitly set in version 4.8.0

**Comment by jcesarmobile** [comment]
This is fixed in Capacitor 4.8.0 and Capacitor 5.0.0 and newer.
It shouldn't require to set `CapacitorCookies` to true, I don't get the error on my sample app anymore after updating to 4.8.0 and it doesn't have `CapacitorCookies` set to true .
For those that it doesn't work unless you add `CapacitorCookies` set to true, please, create a new issue and provide a sample app where it can be reproduced as that shouldn't be needed.
